File: bindings/java-jni/src/main/java/org/maplibre/nativejni/runtime/AmbientCacheOperation.java
package org.maplibre.nativejni.runtime;

/** Ambient cache maintenance operation for a runtime. */
public enum AmbientCacheOperation {
  RESET_DATABASE(1),
  PACK_DATABASE(2),
  INVALIDATE(3),
  CLEAR(4);

  private final int nativeValue;

  AmbientCacheOperation(int nativeValue) {
    this.nativeValue = nativeValue;
  }

  public int nativeValue() {
    return nativeValue;
  }
}

File: bindings/kotlin-native/src/nativeMain/kotlin/org/maplibre/nativeffi/runtime/AmbientCacheOperation.kt
package org.maplibre.nativeffi.runtime

/** Ambient cache maintenance operation for a runtime. */
public enum class AmbientCacheOperation(public val nativeValue: UInt) {
  RESET_DATABASE(1U),
  PACK_DATABASE(2U),
  INVALIDATE(3U),
  CLEAR(4U),
}
